### PR TITLE
Theme showcase: Show 'Beginner' ribbon on some themes

### PR DESCRIPTION
**Pull request by seear**
screen shot 2017-09-29 at 15 19 13

Show a ribbon on themes that are suitable for beginners to help guide theme choice.

pNEWy-9OC-p2

## Testing
* go to /themes
* You should see the new _Beginner_ ribbon on a couple of the themes

/cc @davidakennedy

### CODE REVIEW

**Comment by seear** [comment]
@ockham I moved away from the dedicated selector because it seemed like it was going to involve a function prop [here](https://github.com/Automattic/wp-calypso/blob/49df3a1df796c00317aa1ccac4aa91abfa31bed1/client/my-sites/themes/themes-selection.jsx#L202-L203) which is a bit of an anti-pattern.

**Comment by seear** [comment]
@mapk feel free to style this how you like.

File: client/components/theme/index.jsx [hunk 1]
```diff
@@ -86,6 +88,12 @@ export class Theme extends Component {
 		this.props.onScreenshotClick( this.props.theme.id, this.props.index );
 	};
 
+	isBeginnerTheme = () => {
+		const { theme } = this.props;
+		const skillLevels = get( theme, [ 'taxonomies', 'theme_skill-level' ], null );
+		return !! find( skillLevels, { slug: 'beginner' } );
```
Comment (ockham) [inline client/components/theme/index.jsx]: Isn't lodash's `some` a (more elegant) synonyme for `!! find`?

File: client/components/theme/index.jsx [hunk 2]
```diff
@@ -86,6 +88,12 @@ export class Theme extends Component {
 		this.props.onScreenshotClick( this.props.theme.id, this.props.index );
 	};
 
+	isBeginnerTheme = () => {
+		const { theme } = this.props;
+		const skillLevels = get( theme, [ 'taxonomies', 'theme_skill-level' ], null );
```
Comment (ockham) [inline client/components/theme/index.jsx]: Do we need the `null` fallback?

File: client/components/theme/index.jsx [hunk 3]
```diff
@@ -146,6 +154,7 @@ export class Theme extends Component {
 
 		return (
 			<Card className={ themeClass }>
+				{ this.isBeginnerTheme() && <Ribbon color="green">{ translate( 'BEGINNER' ) }</Ribbon> }
```
Comment (ockham) [inline client/components/theme/index.jsx]: I think I'd rather `translate( 'Beginner' )` and implement the caps via styling. I'll defer to @mapk tho 😄

Comment (mapk) [reply client/components/theme/index.jsx]: I like that idea. Let's default the all caps to the CSS.

**Comment by mapk** [comment]
To get the ribbon to display properly (to extend past the thumbnail and show the triangles of the ribbon folding back behind the thumbnail), we need to make an adjustment to the `.theme` selector and remove the `overflow:hidden;` from it. I did this in the inspector, and it appeared to be fine, but I'm not sure where else the selector is being used.

**Comment by seear** [comment]
`overflow: hidden` was added to fix a few issues in #11229, so we'll need a solution that doesn't break the test cases there.

**Comment by mapk** [comment]
I'm going to get this running locally and test it to find a solution.

**Comment by mapk** [comment]
@johnmaeda has also requested to get ribbons for each of these:

- Beginner
- Intermediate
- Advanced
- Staff Pick

Maybe we can make sure the code is ready for this?

**Comment by seear** [comment]
> Intermediate
Advanced

@kathrynwp Would you be able to add these + descriptions to the `skill-level` taxonomy?

**Comment by kathrynwp** [comment]
> Would you be able to add these + descriptions to the skill-level taxonomy?

Done. Intermediate & Advanced were more challenging to write than Beginner, so I would love input from @Automattic/editorial -

**Intermediate - Explore a collection of themes geared to a moderate skill level.**

**Advanced - Find themes that provide more options and take a bit of extra time to set up.** 

For reference, this is Beginner: **These simple, easy-to-use themes are quick to set up.**

**Comment by seear** [comment]
> Done.

Thanks @kathrynwp

**Comment by michelleweber** [comment]
I think it'd be helpful to go a bit more specific, because users don't know what our defintions of "moderate" and "advanced" are. Maybe something like:

Intermediate - Feel like tinkering? These themes look best with some custom touches and configurations.

Advanced - Custom templates, special features, multi-section home pages -- these themes have more options, and take more time to set up.

**Comment by kathrynwp** [comment]
Thanks @michelleweber - this is great! I've updated Intermediate.

My only concern with the proposed Advanced wording is that I wonder if users might think we're specifically promising custom templates and multi-section homepages, where they are just examples of things they might find in an advanced theme. What do you think of:

Advanced - These themes have more options, such as custom templates and multi-section front pages, and take more time to set up.

**Comment by mapk** [comment]
Visual results of c563684

**Regular implementation**
moved-css

**Vertically stretched image at 400px (#554)**
image-increase-400px-height

**Mobile menu icon still in place (#10661)**
simulator screen shot - iphone x - 2017-10-03 at 14 07 31

**Long names still work (#9700)**
screen shot 2017-10-03 at 2 06 12 pm

**Comment by michelleweber** [comment]
@kathrynwp How about "More of everything: these themes have more options, like custom templates and multi-section front pages, and take more time to set up." It has a teeny bit more voice with a bit of a lead-in.

And if you want to get similarly descriptive with beginner: Simple but beautiful, these themes require little set-up. Low on effort, low on time.

**Comment by johnmaeda** [comment]
We want to be sure to not use WP language as the average person doesn't know what a "custom template" or "multi-section front page" means.

http://Design.blog | @johnmaeda » typed via mobile

**Comment by seear** [comment]
@mapk Nice work on the css 👍 

> Beginner
Intermediate
Advanced
Staff Pick
Maybe we can make sure the code is ready for this?

I'll look at adding these ribbons. Couple of questions:
1) Should we go with the above priority order (if a theme is in more than one of the taxonomies)
2) What color ribbon for _Staff Pick_?

**Comment by michelleweber** [comment]
FWIW, I purposefully used those terms as a shorthand to signal "warning: seriously advanced."

Alt: More of everything: these themes have more options, more settings requiring configuration, and take more time to set up.

**Comment by seear** [comment]
Rebased.

**Comment by seear** [comment]
@mapk think we should add some more preset colors to the ribbon component [here](https://github.com/Automattic/wp-calypso/blob/51d9900ffe2c32a5e341d82ea230bf705d12ae3a/client/components/ribbon/style.scss#L50), or add classes to our theme ribbon and do theme-specific colors for the various taxonomies?

**Comment by seear** [comment]
I propose we ship this PR now to introduce the _Beginner_ ribbons in production, then iterate with a separate PR to add additional ribbons. Any objections?

**Comment by johnmaeda** [comment]
I agree. Let’s start with one. Thanks Steve

**Pull request merged**